lib/puppet/transaction/event.rb

```diff
@@ -94,6 +98,28 @@ def to_yaml_properties
     YAML_ATTRIBUTES & super
   end
 
+  # Calculate and set the corrective_change parameter, based on the old_system_value of the property.
+  # @param [Object] old_system_value system_value from last transaction
+  # @return [bool] true if this is a corrective_change
+  def calculate_corrective_change(old_system_value)
+    # Only idempotent properties, and cases where we have an old system_value
+    # are corrective_changes.
+    @corrective_change = if @property_instance.idempotent? &&
```

Comment:
This combines an `if` expression that has side effects and uses the return value, which is confusing. I would prefer something like this:

```ruby

@corrective_change = @property_instance.idempotent? && !@property_instance.sensitive && !old_system_value.nil?

if @corrective change
  # ...
else
  false
end
```